src/components/button/button.test.tsx

```diff
@@ -0,0 +1,330 @@
+    );
+
+    const button = screen.getByRole("button");
+    expect(button).toHaveStyle("white-space: nowrap");
+  });
+
+  it("renders an anchor element when a custom value is passed to the 'href' prop", () => {
+    render(<Button href="https://www.warnerbros.com/movies/heat">bar</Button>);
+
+    const button = screen.getByRole("link");
+
+    expect(button).toBeVisible();
+  });
+
+  it("sets the 'target' attribute correctly when a custom value is passed to the 'target' prop", () => {
+    render(
+      <Button href="https://www.warnerbros.com/movies/heat" target="_blank">
+        bar
+      </Button>
+    );
+
+    const button = screen.getByRole("link");
+
+    expect(button).toHaveAttribute("target", "_blank");
+  });
+
+  it("sets the 'rel' attribute correctly when a custom value is passed to the 'rel' prop", () => {
+    render(
+      <Button
+        href="https://www.warnerbros.com/movies/heat"
+        rel="noopener noreferrer"
+      >
+        bar
+      </Button>
+    );
+
+    const button = screen.getByRole("link");
+
+    expect(button).toHaveAttribute("rel", "noopener noreferrer");
+  });
+
+  it("renders a tooltip, populated with a custom value that is passed to the 'iconTooltipMessage' prop", async () => {
+    render(<Button iconType="bin" iconTooltipMessage="foo" />);
+
+    const user = userEvent.setup();
+    const button = screen.getByRole("button");
+    await user.hover(button);
+    const tooltip = await screen.findByText("foo");
+
+    expect(tooltip).toBeInTheDocument();
+  });
+
+  it("sets the 'id' attribute correctly when a custom value is passed to the 'id' prop", () => {
+    render(<Button id="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("id", "foo");
+  });
+
+  it("sets the text content of the 'subtext' element when the 'size' prop is 'large' and the 'subtext' prop is passed", () => {
+    render(
+      <Button size="large" subtext="some context about the foo">
+        foo
+      </Button>
+    );
+
+    const subtext = screen.getByText("some context about the foo");
+    expect(subtext).toBeInTheDocument();
+  });
+
+  it("sets the 'className' attribute correctly when a custom value is passed to the 'className' prop", () => {
+    render(<Button className="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveClass("foo");
+  });
+});
+
+describe("events", () => {
+  it("clicks the link on the Button when a 'href' is passed and the space key is pressed", async () => {
+    render(<Button href="https://www.warnerbros.com/movies/heat">bar</Button>);
+
+    const button = screen.getByRole("link");
+    const clickMock = jest.fn();
+
+    button.addEventListener("click", clickMock);
+    fireEvent.keyDown(button, { key: " " });
+
+    expect(clickMock).toHaveBeenCalledTimes(1);
+  });
+
+  it("does not click the link on the Button when a 'href' is passed and any other key is pressed", () => {
```

Comment:
We're checking on this one that a `onClick` is only called when the button is a link and a space key is pressed, this test is just checking it's not called when a random key is pressed. Just for this example I chose the enter key, but in theory it could be any key

Reply:
Happy for suggestions on the description, though, as it's not my forte`